drivers/flash/soc_flash_numaker.c

```diff
@@ -0,0 +1,290 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ *  Copyright (c) 2023 Nuvoton Technology Corporation.
+ */
+
+#define DT_DRV_COMPAT nuvoton_numaker_fmc
+
+#include <string.h>
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include "flash_priv.h"
+#include <NuMicro.h>
+
+LOG_MODULE_REGISTER(flash_numaker, CONFIG_FLASH_LOG_LEVEL);
+
+#define SOC_NV_FLASH_NODE DT_INST(0, soc_nv_flash)
+
+extern int32_t FMC_EraseBank(uint32_t u32_bank_addr);
```

Comment:
Yes, hal should be the better choice.
I notified the owner of exernal BSP/SDK to resolve it.
So, current alternative solution is to add this declaration to avoid compiler warning.

Reply:
Could you open a PR for the HAL directly?

Reply:
Yes, I could create one PR for HAL by Tomorrow. I will also try to ask external bsp/sdk .